### PR TITLE
Target Selector

### PR DESCRIPTION
**Pull request by ubeF**
Target Selector used for sabotage and certain action
- Caller can select any player except himself or players with a "sabotage free card"

### CODE REVIEW

File: select_target/select_target.gd [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+func display(player: Player):
+	print("Choosing target")
+	self.player = player
+	
+	for target in State.players:
+		if target != player:
+			if target.sabotage_card:
+				target.sabotage_card = false
+			else: 
+				var button = preload("res://sabotage_field/target_button.tscn").instantiate()
```
Comment (hfxbse) [inline select_target/select_target.gd]: The file seems to have been moved since writing this.

File: select_target/select_target.gd [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+				$GridContainer.add_child(button)
```
Comment (hfxbse) [inline select_target/select_target.gd]: Nodes from scenes should be selected only once with `@onready` to improve maintainability.

**Review by hfxbse — CHANGES_REQUESTED** [review]
Not sure how to test this, as it seems to be not connected to anything right now. Maybe consider it a draft till then?

**Comment by ubeF** [comment]
> Not sure how to test this, as it seems to be not connected to anything right now. Maybe consider it a draft till then?

I tested it in #35  but that branch is still waiting for the implementation of duels and i need the target selector for the action fields. So i separated it.

I could merge this branch into action field but i fear it might get ugly

**Review by hfxbse — APPROVED** [review]

**Comment by hfxbse** [comment]
> > Not sure how to test this, as it seems to be not connected to anything right now. Maybe consider it a draft till then?
> 
> I tested it in #35 but that branch is still waiting for the implementation of duels and i need the target selector for the action fields. So i separated it.
> 
> I could merge this branch into action field but i fear it might get ugly

I guess it is working in #35 but I do not know what the wanted behavior is exactly, so I have to trust you on that one.

The player icons seem to be slightly misaligned to the left, but that is all.

**Pull request merged**